### PR TITLE
Miscellaneous improvements and bug fixes

### PR DESCRIPTION
**Pull request by Marton-Zeisler**
Hi everyone,

I cloned the main repository and did some fixes and added some new features to the app.

Video of the app showing the changes: [Video Link](https://drive.google.com/file/d/1Y3WjxmgwNlPpNNy6pXH1Otql8eNMrEVW/view?usp=sharing)

**Features Added:**
- Improved the UI and UX of the app to give it a more modern and cleaner look and feel
- Added a tab bar navigation system to help the user find the main screens of the app more easier. Tab bars are known are for its simplicity and ease of use, lots of popular apps use it

**Login Screen:**
- Added a touch gesture recogniser on the screen so the keyboard can be dismissed when the user touches somewhere else on the screen during username or password input
- Error message appears for both textfields at the same time if both textfields are empty
- Added a show/hide button to the password field to enable the user to see the password in plain text too
- Tapping on the return key(Next) in the username textfield will make the password textfield active and the return key(Done) in the password will trigger the sign in button action
- Landscape support added, the interface looks nice even in landscape mode

**Servers Screen:**
- Replaced table view with a collection view and added server icons to improve the user interface design
- Fixed a bug where the refresh indicator wouldn't change to its tint colour when the user first loads the servers screen

**Downlads Screen:**
- Keyboard now gets dismissed if the user touches outside the search bar during search input
- Increased width for the segmented control makes the interface look better

**Settings Screen:**
- Fixed the order of the sign out action buttons. Usually apps put the delete button on the left side and the delete(main action) on the right side
- Added a selection colour to each field to indicate which settings item was pressed by the user
- Removed unnecessary pop up when no email account is setup and user tries to send feedback. iOS already displays a pop up warning
- Fixed a bug where the username and password fields would not be empty after signing out when the user signed in previously in the same session

**Settings Connection Screen:**
- Fixed a crash which would occur if the user tried to change the connection type without going into a HDA previously on the HDA screen
- Changed the back button colour to white from blue to match the black-white theme of the app

### CODE REVIEW

File: AmahiAnywhere/AmahiAnywhere.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -783,14 +787,14 @@
...
-				DEVELOPMENT_TEAM = 325LAYFA25;
+				DEVELOPMENT_TEAM = 83F32NLU7E;
 				ENABLE_BITCODE = NO;
 				INFOPLIST_FILE = AmahiAnywhere/Info.plist;
 				LD_RUNPATH_SEARCH_PATHS = (
 					"$(inherited)",
 					"@executable_path/Frameworks",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = com.amahi.AmahiAnywhere;
+				PRODUCT_BUNDLE_IDENTIFIER = com.amahi.AmahiAnywhere.Marton.Test;
```
Comment (magauran) [inline AmahiAnywhere/AmahiAnywhere.xcodeproj/project.pbxproj]: Hi, @Marton-Zeisler! You forgot to bring back the old Bundle ID.

Comment (Marton-Zeisler) [reply AmahiAnywhere/AmahiAnywhere.xcodeproj/project.pbxproj]: Oh yes, apologies for that

**Comment by cpg** [comment]
Would you please consider rebasing against `beta` and split it in a handful of major PRs?

Thanks

**Comment by Marton-Zeisler** [comment]
Yeah I can do that

**Comment by Marton-Zeisler** [comment]
Tried to clone the beta repository but can't even build the project due to errors. I think the last commit contains errors that weren't fixed. I posted screenshots under the beta's last commit:

[Link to the last commit in the beta branch](https://github.com/amahi/ios/commit/1449d561a688032ad0e5a6738d20c0f81d290e6a)

**Comment by iswiftshek** [comment]
What errors are you getting? @Marton-Zeisler

**Comment by codedentwickler** [comment]
@Marton-Zeisler , I think the issue you are facing is in the ApiConfig file you got is the old file. The properties has been changed, for example, `BASE_URL` was changed to `baseUrl`.

**Comment by Marton-Zeisler** [comment]
Okay thank you @codedentwickler, that fixed one of the files. The only error I get now is this:

Maybe the pod version of SDWebImageManager that I use is different than yours and one of its functions requires additional arguments. Can you tell me which version you use for this? I just ran pod update to get the latest version of everything

Screenshot 2019-04-11 at 11 32 31

**Comment by codedentwickler** [comment]
Yes about that. I think you did `pod update` instead of `pod install`. Checkout the `beta` branch and run `pod install`. The current version on beta is `4.4.6 `, you can find it here. https://github.com/amahi/ios/blob/beta/AmahiAnywhere/Podfile.lock

**Comment by Marton-Zeisler** [comment]
Okay it builds now. Thank you

**Comment by Marton-Zeisler** [comment]
@cpg, so Carlos, is there any features that you don't like? For example do you like the tab bar based navigation system?

Just so I know which features to add to the beta branch

**Comment by Marton-Zeisler** [comment]
I created the first pull request for the beta branch, that pull request only includes the fixes and improvements to the Logins screen. I'll work on new pull requests soon.

**Comment by cpg** [comment]
Are we to close this one so that we do things in piecemeal fashion?

**Comment by Marton-Zeisler** [comment]
> Are we to close this one so that we do things in piecemeal fashion?

I still use it for reference when I work on the small pieces to put it together so let’s keep it if possible and then I can close it when I no longer need it.

**Pull request closed without merging**